### PR TITLE
feat: Introduce gradio for creating interactive and visual demo

### PR DESCRIPTION
**Pull request by simon824**
```
export PYTHONPATH=${PROJECT_ROOT_DIR}/hugegraph-llm/src:${PROJECT_ROOT_DIR}/hugegraph-python-client/src
python3 ./hugegraph-llm/src/hugegraph_llm/utils/gradio_demo.py
```

image
image
image

### CODE REVIEW

File: hugegraph-llm/src/hugegraph_llm/operators/kg_construction_task.py [hunk 1]
```diff
@@ -58,7 +60,8 @@ def print_result(self):
         self.operators.append(PrintResult())
         return self
 
-    def run(self):
-        result = ""
+    def run(self) -> Dict[str, Any]:
+        context = ""
```
Comment (vichayturen) [inline hugegraph-llm/src/hugegraph_llm/operators/kg_construction_task.py:64]: `context=None` may be better.

File: hugegraph-llm/src/hugegraph_llm/utils/gradio_demo.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+def build_kg(text, schema, disambiguate_word_sense, commit_to_hugegraph):
+    builder = KgBuilder(LLMs().get_llm())
+    # 判断 schema 是不是json结构
+    if schema:
+        try:
+            schema = json.loads(schema.strip())
+            builder.import_schema(from_user_defined=schema)
+        except json.JSONDecodeError as e:
+            print(e)
+            builder.import_schema(from_hugegraph=schema)
```
Comment (vichayturen) [inline hugegraph-llm/src/hugegraph_llm/utils/gradio_demo.py:97]: When we want to import schema from huge graph, we input the graph name as `schema`.
However, `schema` is `str` can be `json.loads` without exception.

Comment (simon824) [reply hugegraph-llm/src/hugegraph_llm/utils/gradio_demo.py:97]: Importing from hugegraph only needs to specify graph_name, which is not a json structure. Of course, this design is not elegant, and we can find ways to optimize it later.
- from_user_defined: {"xx": "xx"}
- from_hugegraph: "tmp_graph"

File: hugegraph-llm/examples/build_kg_test.py [hunk 3]
```diff
@@ -49,13 +49,15 @@
     }
 
     (
-        builder.import_schema(from_hugegraph="xxx")
+        builder
+        # .import_schema(from_hugegraph="xxx")
         .print_result()
```
Comment (liuxiaocs7) [inline hugegraph-llm/examples/build_kg_test.py:54]: need comment?

File: hugegraph-llm/src/hugegraph_llm/utils/gradio_demo.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+        .query_graph_for_rag()
+        .synthesize_answer()
+        .run(verbose=True)
+    )
+    return res
+
+
+def build_kg(text, schema, disambiguate_word_sense, commit_to_hugegraph):
+    builder = KgBuilder(LLMs().get_llm())
+    # 判断 schema 是不是json结构
```
Comment (liuxiaocs7) [inline hugegraph-llm/src/hugegraph_llm/utils/gradio_demo.py:90]: remove

**Review by liuxiaocs7 — APPROVED** [review]

**Pull request merged**